File: Practica-8/ABC.java
package papa;

class A{
    public int x;
    public int z;
    public A(int x, int z){
        this.x = x;
        this.z = z;
    }
    public void incrementaXZ(){
        this.x++;
        this.z++;
    }
    public void incrementaZ(){
        this.z++;
    }
}
class B{
    public int y;
    public int z;
    public B(int y, int z){
        this.y = y;
        this.z = z;
    }
    public void incrementaYZ(){
        this.y++;
        this.z++;
    }
    public void incrementaZ(){
        this.z++;
    }
}
class D{
    A a;
    B b;
    public D(int x, int y, int z){
        a = new A(x,z);
        b = new B(y,z);
    }
    public void incrementaXYZ(){
        a.x++;
        b.y++;
        a.z++;
    }
    /*
    public void Mostrar(){
        System.out.println("x = " + a.x + ", y = " + b.y + ", z = " + a.z);
    }
    */
}


public class ABC {
    
    public static void main(String[] args) {
        D cal = new D(1,2,3);
        System.out.println("x = " + cal.a.x + ", y = " + cal.b.y + ", z = " + cal.a.z);
        cal.incrementaXYZ();
        //
        cal.a.incrementaXZ();
        //
        cal.b.incrementaYZ();
        //
        cal.a.incrementaZ();
        System.out.println("x = " + cal.a.x + " y = "+ cal.b.y + " z = "+ cal.a.z);
    }
    
}

File: Practica-8/ABC.py
class A:
	def __init__(self, x, z):
		self.x = x
		self.z = z
	def incrementaXZ(self):
		self.x += 1
		self.z += 1
	def incrementaZ(self):
		self.z += 1
	
class B:
	def __init__(self,y,z):
		self.y = y
		self.z = z
	def incrementaYZ(self):
		self.y += 1
		self.z += 1
	def incrementaZ(self):
		self.z += 1
		
class D(A, B):
	def __init__(self, x, y, z):
		A.__init__(self,x,z)
		B.__init__(self,y,z)
	def incrementaXYZ(self):
		self.x += 1
		self.y += 1
		self.z += 1
		
cal = D(1,2,3)
print(f"x,y,z = {cal.x}, {cal.y}, {cal.z}")

print("----------------")

cal.incrementaXZ()
cal.incrementaZ()
cal.incrementaXYZ()
print(f"x,y,z = {cal.x}, {cal.y}, {cal.z}")